homeassistant/helpers/entity.py

```diff
@@ -314,6 +314,13 @@ def name(self) -> str | None:
             return self._attr_name
         if hasattr(self, "entity_description"):
             return self.entity_description.name
+        if self.translation_key is not None:
+            assert self.platform
```

Comment:
```suggestion
        if self.platform and self.translation_key is not None:
```

Reply:
There's no need to check this IMHO, `Entity.platform` can't be `None` once the entity has been added to Home Assistant. I think we should either type `Entity.platform` in a similar way as we type `Entity.hass`: https://github.com/home-assistant/core/blob/23823204fc9609fdb210981338b2cd2c4f050c4a/homeassistant/helpers/entity.py#L238

Or, as another option, only add the `platform` attribute when the Entity is added and remove it when it's removed: https://github.com/home-assistant/core/pull/88321

I'm not sure if the 2nd option is quite worth the squeeze though.